src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -28,20 +29,19 @@ internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
         public static bool IsAspNetCore20 = typeof(WebHostBuilder).GetTypeInfo().Assembly.GetName().Version.Major >= 2;
 
         private readonly TelemetryClient client;
-        private readonly ICorrelationIdLookupHelper correlationIdLookupHelper;
-        private readonly string sdkVersion;
+        private readonly IApplicationIdProvider applicationIdProvider;
+        private readonly string sdkVersion = SdkVersionUtils.GetVersion();
         private const string ActivityCreatedByHostingDiagnosticListener = "ActivityCreatedByHostingDiagnosticListener";
 
         /// <summary>
         /// Initializes a new instance of the <see cref="T:HostingDiagnosticListener"/> class.
         /// </summary>
         /// <param name="client"><see cref="TelemetryClient"/> to post traces to.</param>
-        /// <param name="correlationIdLookupHelper">A store for correlation ids that we don't have to query it everytime.</param>
-        public HostingDiagnosticListener(TelemetryClient client, ICorrelationIdLookupHelper correlationIdLookupHelper)
+        /// <param name="applicationIdProvider">Nullable Provider for resolving application Id to be used by Correlation.</param>
+        public HostingDiagnosticListener(TelemetryClient client, IApplicationIdProvider applicationIdProvider)
```

Comment:
Inject TelemetryConfiguration here, and obtain appid provider from the Configuration, so that even if there is no AppId provider(occurs if user removes it from DI), this wouldn't result in exception.

Reply:
@cijothomas  Are you sure we want to do that?
My thought is that only the NetFramework Modules will be expecting a TelemetryConfiguration.
All of your NetCore Modules are using DependencyInjection, and it would make sense to keep that consistent. 

As an alternative; I can make this parameter null by default or make a second ctor without the IApplicationIdProvider parameter, to support DependencyInjection.


I assume consistency would be prefered, but I will defer to your judgement here :)

If I make changes here, would you like me to make the same changes in `OperationCorrelationTelemetryInitializer`?

Reply:
For me it’s about somebody forgot to add it, not somebody removed it. In some custom scenarios.

Reply:
@SergeyKanzhelev, But the TelemetryConfiguration is also being constructed by DependencyInjection. 
Maybe I'm not understanding a user custom config scenario. :)

Would something like this be an appropriate alternative...
```
public HostingDiagnosticListener(TelemetryClient client, IApplicationIdProvider applicationIdProvider, TelemetryConfiguration telemetryConfiguration)
{
     this.applicationIdProvider = applicationIdProvider ?? telemetryConfiguration.ApplicationIdProvider;
}
```

Reply:
@cijothomas explained this to me offline. I'm cleaning this up now.
Thanks